Add files via upload

ArmMart0896 opened the pull request:
Sprint_4_Martirosyan

ArmMart0896 committed:
Add files via upload

Sprint_4_Martirosyan

alesyaisachenko commented on Sprint_4/Sprint_4_Martirosyan/src/main/java/pages/OrderPage.java:
```diff
@@ -0,0 +1,110 @@
+
+    // Окно для кого самокат
+    private final By orderTitle = By.className("Order_Header__BZXOb");//заголовок
+    private By windowOrderForWho = By.xpath("/html/body/div/div/div[2]");
```
Это нужно обязательно доработать:

Это плохой вариант локатора поиска, локатор должен находить элемент за 2-3 шага и привязан быть к уникальным характеристикам элемента.


Доработай, плиз, все аналогичные локаторы в этом классе

alesyaisachenko commented on Sprint_4/Sprint_4_Martirosyan/src/main/java/pages/OrderPage.java:
```diff
@@ -0,0 +1,110 @@
+        return this;
+    }
+    public OrderPage coockiesBtn() {
+        driver.findElement(By.xpath("/html/body/div/div/div[3]/div/div[2]/button")).click();
```
Это нужно обязательно доработать:

By.xpath("/html/body/div/div/div[3]/div/div[2]/button") -- выдели, плиз, отдельную переменную для этого элемента

alesyaisachenko commented on Sprint_4/Sprint_4_Martirosyan/src/test/java/TestImportantQuestions.java:
```diff
@@ -0,0 +1,52 @@
+
+@RunWith(Parameterized.class)
+public class TestImportantQuestions {
+    public final int questIndex;
```
Можно лучше:

лучше модификатор доступа private

alesyaisachenko commented on Sprint_4/Sprint_4_Martirosyan/src/test/java/TestPositiveOrder.java:
```diff
@@ -0,0 +1,95 @@
+                orderPage.inputDeliveryMessage(message);
+                orderPage.clickLastOrderBtn();
+                orderPage.clickYesBtn();
+                orderPage.clickCheckOrderStatus();
```
Это нужно исправить:
сейчас тест падает на этом шаге (запусти тест на firefox и ты это увидишь)-- это нужно обязательно пофиксить.

P.s. проще сделать проверку, что появляется сообщение с текстом "Заказ оформлен"

alesyaisachenko commented on Sprint_4/Sprint_4_Martirosyan/src/test/java/TestFromLesson.java:
```diff
@@ -0,0 +1,38 @@
+public class TestFromLesson {
+    private WebDriver driver;
+    private final String incorrectOrderNum = "5";
+    @Before
```
Это нужно исправить:


блоки @Before и @After у тебя дублируются в каждом тесте.


Дублирование кода нужно избегать.


Как это можно сделать:


Создать класс BaseTest, в него вынести блоки @before и @after. Остальные тестовые классы наследовать от BaseTest.

ArmMart0896 committed:
Исправил в соответствии с замечаниями

1) Исправил локаторы в orderPage;
2) Выделил отдельную переменную для кнопки кукис;
3) Изменил модификатор доступа на private в тесте TestImportantQuestions;
4) добавил проверку текста "Заказ Оформлен" в сообщении;
5) Вынес аннотации @Before и @After в отдельный класс
